File: server-spring-boot/src/main/java/com/finflow/server/entity/Merchant.java
package com.finflow.server.entity;

import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Merchant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String businessName;
    private String email;
    private String phone;
    private String password;
    private String businessType; // "Electricity", "Water", "Gas"
    private String status;

    @OneToMany(mappedBy = "merchant")
    private List<Invoice> invoices;
}

File: server-dot-net/Model/Merchant.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server_dot_net.Model
{
    [Table("merchant")]
    public class Merchant
    {
        [Key]
        public long Id { get; set; }

        [Column("business_name")]
        [Required]
        public string BusinessName { get; set; }

        [Required]
        public string Email { get; set; }

        public string Phone { get; set; }

        [Required]
        public string Password { get; set; }

        [Column("business_type")]
        public string BusinessType { get; set; } // Electricity, Water, Gas

        public string Status { get; set; }

        public ICollection<Invoice> Invoices { get; set; }
    }
}
